tsinfer/inference.py

```diff
@@ -714,6 +715,9 @@ def add_sites(self, exclude_positions=None):
                     # Time = freq of *all* derived alleles. Note that if n_alleles > 2 this
                     # may not be sensible: https://github.com/tskit-dev/tsinfer/issues/228
                     time = counts.derived / counts.known
+                if np.isnan(time):
```

Comment:
Should be ``elif`` here - need more tests!

Reply:
I'm not sure it should be, should it. If it's `tskit.UNKNOWN_TIME` then it get replaced with `time=freq` on the line above, so it shouldn't be excluded from inference on this line. The only reason I didn't use an elif is that this catches the case when `counts.known` is 0, which seems worth doing (even though it's actually excluded by the logic above, which I guess could therefore be simplified). So I think the tests *do* cover this?

Reply:
But ``use_site`` will be False in all cases where the time is a NaN by the time we get to line 720, won't it?

Reply:
Ah, right, I see, you've set the time within the condition block. Sorry, my bad.